### PR TITLE
Update gosdk to support enableSSMAcess

### PR DESCRIPTION
**Pull request by ananth102**
Issue #, if available:

Description of changes:
`generator.yaml` - Added various shapes/fields to ignore
`go.mod` - Updated go sdk
`go.sum` - Artifacts of go mod tidy
`test/scripts/install-mockery.sh` - Updating mockery for unit test
`test/mocks/aws-sdk-go/sagemaker/SageMakerAPI.go` - Updating Sagemaker mock to support new apis

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: generator.yaml [hunk 1]
```diff
@@ -1027,8 +1032,29 @@ ignore:
       - ExplainerConfig
       - HyperParameterTuningJobStrategyConfig
       - DisableProfiler
+      - BatchTransformInput_
```
[the remark this answers is not in the document]
Comment (ananth102) [reply generator.yaml:1035]: Yes, that shape has a trailing underscore in the go sdk. https://docs.aws.amazon.com/sdk-for-go/api/service/sagemaker/#BatchTransformInput_

**Comment by surajkota** [comment]
/approve

Lets add this parameter to one of the existing unit tests?

File: generator.yaml [hunk 2]
```diff
@@ -1027,8 +1032,29 @@ ignore:
...
+      - MonitoringParquetDatasetFormat
+      - SpaceName
+      - DefaultSpaceSettings
+      - BatchDataCaptureConfig
+      - CodeRepositories
+      - TableFormat
+      - RandomSeed
+      - TrainingImageConfig
+      - ShadowProductionVariants
```
Comment (surajkota) [inline generator.yaml:1044]: lots of backlog :(

**Pull request merged**